src/fidesops/ops/service/email/email_dispatch_service.py

```diff
@@ -16,30 +16,64 @@
     EmailServiceType,
     SubjectIdentityVerificationBodyParams,
 )
+from fidesops.ops.tasks import DatabaseTask, celery_app
 from fidesops.ops.util.logger import Pii
 
 logger = logging.getLogger(__name__)
 
 
+class InvalidBodyParams(Exception):
+    """An exception thrown when passed email body data does not match an allowed schema"""
+
+
+def _validate_email_body_params(
+    email_body_params: Dict[str, Union[str, int]]
+) -> SubjectIdentityVerificationBodyParams:
+    """Raises an exception if `email_body_params` doesn't fit one of the allowed schemas"""
+    email_body_schema_allowlist = [SubjectIdentityVerificationBodyParams]
+    for schema in email_body_schema_allowlist:
+        try:
+            # If the schema is valid, exit early
+            return schema.parse_obj(email_body_params)
+        except ValidationError:
+            continue
+
+    allowed_schemas_str = ",".join(str(email_body_schema_allowlist))
+    raise InvalidBodyParams(
+        f"`email_body_params` must match one of {allowed_schemas_str}"
+    )
+
+
+@celery_app.task(base=DatabaseTask, bind=True)
+def dispatch_email_task(
+    self: DatabaseTask,
+    action_type: EmailActionType,
+    to_email: str,
+    email_body_params: Dict[str, Union[str, int]],
+) -> None:
+    """
+    A wrapper function to dispatch an email task into the Celery queues
+    """
+    valid_params_schema = _validate_email_body_params(email_body_params)
+    with self.session as db:
+        dispatch_email(
+            db,
+            action_type,
+            to_email,
+            valid_params_schema,
+        )
+
+
 def dispatch_email(
     db: Session,
     action_type: EmailActionType,
     to_email: Optional[str],
-    email_body_params: Union[SubjectIdentityVerificationBodyParams],
+    email_body_params: SubjectIdentityVerificationBodyParams,
```

Comment:
My work to add another email type in https://github.com/ethyca/fidesops/pull/1246/files adds another type to this Union, I don't know if we need to get rid of this. Just more places to reconcile merges.